mesonbuild/linkers/detect.py

```diff
@@ -89,10 +90,6 @@ def guess_win_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
                 for_machine, override, exelist=compiler, prefix=comp_class.LINKER_PREFIX,
                 version=search_version(o), direct=False, machine=None)
 
-    if value is not None and invoked_directly:
-        compiler = value
-        # We've already hanedled the non-direct case above
-
```

Comment:
Same here, now we're looking for linkers that are invoked directly, not through the compiler driver. The more I liok at this, the more I feel ilke this change is wrong.

Reply:
If I explicitly tell meson to use the linker directly it should shut up and do as I say!
meson really tries too hard to be clever .... (personally I think the whole stuff should be rewritten. I saw many fragments of code duplication, especially for defaults in complier/detect.py and linker/detect.py )

Reply:
As far as I'm concerned, meson _has_ to be clever - if you "just use the linker directly" how is it going to know what flags to use for linking? Path conversion, library search locations, whether the target is static or shared etc... There is no concept of a dumb linker because it isn't sensible, I think.

One of my biggest pain points with meson is also one of its biggest strengths - if your compiler is not known, you are forced to implement it properly so _all_ future consumers of that compiler benefit from correct detection.